cmd/tests/cmd_test.go

```diff
@@ -22,160 +22,187 @@ import (
 	"github.com/stretchr/testify/require"
 	"github.com/tidwall/gjson"
 	"go.k6.io/k6/cloudapi"
+	"go.k6.io/k6/cmd"
+	"go.k6.io/k6/cmd/state"
 	"go.k6.io/k6/errext/exitcodes"
 	"go.k6.io/k6/lib"
 	"go.k6.io/k6/lib/consts"
 	"go.k6.io/k6/lib/testutils"
 	"go.k6.io/k6/lib/testutils/httpmultibin"
 )
 
+func TestDeprecatedOptionWarning(t *testing.T) {
+	t.Parallel()
+
+	ts := state.NewGlobalTestState(t)
+	ts.CmdArgs = []string{"k6", "--logformat", "json", "run", "-"}
+	ts.Stdin = bytes.NewBuffer([]byte(`
+		console.log('foo');
+		export default function() { console.log('bar'); };
+	`))
+
+	cmd.ExecuteWithGlobalState(ts.GlobalState)
+
+	logMsgs := ts.LoggerHook.Drain()
+	assert.True(t, testutils.LogContains(logMsgs, logrus.InfoLevel, "foo"))
+	assert.True(t, testutils.LogContains(logMsgs, logrus.InfoLevel, "bar"))
+	assert.Contains(t, ts.Stderr.String(), `"level":"info","msg":"foo","source":"console"`)
+	assert.Contains(t, ts.Stderr.String(), `"level":"info","msg":"bar","source":"console"`)
+
+	// TODO: after we get rid of cobra, actually emit this message to stderr
+	// and, ideally, through the log, not just print it...
+	assert.False(t, testutils.LogContains(logMsgs, logrus.InfoLevel, "logformat"))
+	assert.Contains(t, ts.Stdout.String(), `--logformat has been deprecated`)
+}
+
 func TestVersion(t *testing.T) {
 	t.Parallel()
 
-	ts := newGlobalTestState(t)
-	ts.args = []string{"k6", "version"}
-	newRootCommand(ts.globalState).execute()
+	ts := state.NewGlobalTestState(t)
+	ts.CmdArgs = []string{"k6", "version"}
+	cmd.ExecuteWithGlobalState(ts.GlobalState)
 
-	stdOut := ts.stdOut.String()
```

Comment:
Is there a reason to rename `stdOut` to `stdout` everywhere? If this was new code, I won't care too much one way or the other, but in this PR, a big chunk of huge diff is just this one change and it's distracting, so my vote would be to keep `stdOut` unchanged

Reply:
Yeah I highly prefer to not mass rename variable/names at the same time you are doing actual refactoring where you move stuff around. Especially in this big diffs.

Renaming the thing that is moved is usually okay, but other stuff usually just make the changes harder to follow.

The thing *I* had problems was the rename of `testState` to `ts` - yes it is more consistent, but it does introduce a change *everywhere* it is used even in the places nothing else changes. And it is used quite a lot ;)

Reply:
This PR standardizes the capitalization of `GlobalState.Std{out,err}` to match `os.Std{out,err}`, so making local vars match that just made sense. I didn't do an exhaustive job changing it everywhere, but I would still prefer to keep it as is, instead of doing even more work now to revert it back to an IMO awkward capitalization.

Same with `s/testState/ts/`, and `s/globalState/gs/`. Previously we used both, and this was done to standardize things.

It's not unbearably distracting, right? After all, refactoring is mostly renaming things to begin with... :smile:

But I'll keep your comments in mind for future PRs, and do these in separate commits.

Reply:
But these are not `GlobalState.Stdout`, this is a local variable in a whole bunch of tests :confused:

Reply:
Right. I made the local variable match the lowercase change of `{out,err}` of the `GlobalState` fields. `stdOut := ts.Stdout` would look weird otherwise.

I feel like we're bikeshedding at this point. :sweat_smile: If you feel strongly about this, I can change it back, but like I said, it's more work for something that doesn't make sense.

Reply:
Fair enough, it's not that big of a big deal, but it makes the PR review more difficult for no apparent benefit. I am usually on the other side of these discussions, since I am the one who usually made huge PRs (though I am trying to be better!), so I'll shut up this time :sweat_smile:

Reply:
Alright, I'll keep this in mind for future PRs, and do the noisy renames in separate commits.